Compiler error for openxr module on MacOS when compiling without Vulkan

### Tested versions

Reproduced in 4.5 dev 931820d33cbc38328e4014d65dd62c359b5116fc (master branch as of submitting)

The code that errors is introduced by: https://github.com/godotengine/godot/pull/99768
But I suspect the actual cause may lie somewhere else.

### System information

Godot v4.5.dev (931820d33) - macOS Sequoia (15.4.1) - Multi-window, 1 monitor - Metal (Forward+) - integrated Apple M4 (Apple9) - Apple M4 (10 threads)

### Issue description

Compiling the Godot editor without Vulkan causes a vulkan only if statement to be executed.

```
if (rendering_driver == "vulkan") {
		request_extensions[XR_FB_FOVEATION_VULKAN_EXTENSION_NAME] = &fb_foveation_vulkan_ext;
}
```
Causing this error:
```
modules/openxr/extensions/openxr_fb_foveation_extension.cpp:76:22: error: use of undeclared identifier 'XR_FB_FOVEATION_VULKAN_EXTENSION_NAME'
   76 |                 request_extensions[XR_FB_FOVEATION_VULKAN_EXTENSION_NAME] = &fb_foveation_vulkan_ext;
      |        
```

### Steps to reproduce

`scons vulkan=no platform=macos arch=arm64`

### Minimal reproduction project (MRP)

N/A

--- a/modules/openxr/extensions/openxr_fb_foveation_extension.cpp
+++ b/modules/openxr/extensions/openxr_fb_foveation_extension.cpp
@@ -72,9 +72,11 @@ HashMap<String, bool *> OpenXRFBFoveationExtension::get_requested_extensions() {
 	request_extensions[XR_FB_FOVEATION_EXTENSION_NAME] = &fb_foveation_ext;
 	request_extensions[XR_FB_FOVEATION_CONFIGURATION_EXTENSION_NAME] = &fb_foveation_configuration_ext;
 
+#ifdef XR_USE_GRAPHICS_API_VULKAN
 	if (rendering_driver == "vulkan") {
 		request_extensions[XR_FB_FOVEATION_VULKAN_EXTENSION_NAME] = &fb_foveation_vulkan_ext;
 	}
+#endif // XR_USE_GRAPHICS_API_VULKAN
 
 	return request_extensions;
 }
@@ -97,9 +99,11 @@ void OpenXRFBFoveationExtension::on_instance_destroyed() {
 
 bool OpenXRFBFoveationExtension::is_enabled() const {
 	bool enabled = swapchain_update_state_ext != nullptr && swapchain_update_state_ext->is_enabled() && fb_foveation_ext && fb_foveation_configuration_ext;
+#ifdef XR_USE_GRAPHICS_API_VULKAN
 	if (rendering_driver == "vulkan") {
 		enabled = enabled && fb_foveation_vulkan_ext;
 	}
+#endif // XR_USE_GRAPHICS_API_VULKAN
 	return enabled;
 }
 

